### DISCUSSION TITLE
`@turbo` and destructuring assignments

### DISCUSSION THREAD

**Discussion by matthias314** [JuliaSIMD/LoopVectorization.jl#421] [General]
It seems that destructuring assignments inside a loop may lead to wrong results. The example
```
function swap(v, w)
    @turbo for i in axes(v, 1)
        v[i], w[i] = w[i], v[i]
    end
end
```
gives
```
julia> v = [0, 0];
julia> w = [1, 1];
julia> swap(v, w);
julia> v[1], w[1]
(1, 1)     # wrong
```
I don't know if this is a bug or part of the "don't rely on a specific execution order" rule. In the latter case, would it make sense to mention this case explicitly in the documentation? After all, it's formally a single statement.

**Comment by chriselrod** [comment]
It's a bug.
